src/channels.rs

```diff
@@ -240,6 +326,34 @@ impl Course {
         guild.delete_role(&ctx.http, self.role).await?;
         Ok(())
     }
+
+    async fn rename_course(
+        &self,
+        new_name: &str,
+        ctx: &Context,
+        guild: GuildId,
+    ) -> serenity::Result<()> {
+        for channel in &self.channels {
+            match channel.to_channel(&ctx.http).await? {
+                SerenityChannel::Guild(mut channel) => {
+                    let prefix_index = channel.name.find("-").unwrap_or(channel.name.len());
+                    let prefix = &channel.name[..prefix_index].to_string();
```

Comment:
unnecessary `to_string()` call here

Reply:
Without allocating, I'm not sure what's the best way to solve this:
```rs
error[E0502]: cannot borrow `channel` as mutable because it is also borrowed as immutable
   --> src/channels.rs:341:21
    |
340 |                       let prefix = &channel.name[..prefix_index];
    |                                     ------------ immutable borrow occurs here
341 | /                     channel
342 | |                         .edit(&ctx.http, |c| c.name(format!("{}-{}", &prefix, new_name)))
    | |__________________________----___________________________________________________________^ mutable borrow occurs here
    |                            |
    |                            immutable borrow later used by call
```